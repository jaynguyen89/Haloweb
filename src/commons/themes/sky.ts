import { createTheme, Shadows } from '@mui/material';
import _cloneDeep from 'lodash/cloneDeep';
import { DEFAULT_THEME_SHADOWS } from '../constants';
import colors from '../colors';

const shadows = _cloneDeep(DEFAULT_THEME_SHADOWS);
shadows[0] = `0 3px 6px rgba(0,0,0,0.16)`;
shadows[1] = `0 3px 6px rgba(0,0,0,0.23)`;

export const ThemeSky = createTheme({
    palette: {
        text: {
            primary: colors.white,
            secondary: colors.lightGrey,
            disabled: colors.darkGrey,
        },
        primary: {
            main: colors.lightBlack,
            light: colors.darkerGrey,
            dark: colors.black,
            contrastText: colors.lightBlack,
        },
        secondary: {
            main: colors.darkerTeal,
            light: colors.lighterTeal,
            dark: colors.deepTeal,
            contrastText: colors.teal,
        },
        error: {
            main: colors.darkerRed,
            light: colors.lighterRed,
            dark: colors.deepRed,
            contrastText: colors.red,
        },
        warning: {
            main: colors.darkerYellow,
            light: colors.lighterYellow,
            dark: colors.deepYellow,
            contrastText: colors.yellow,
        },
        info: {
            main: colors.darkerBlue,
            light: colors.lighterBlue,
            dark: colors.deepBlue,
            contrastText: colors.blue,
        },
        success: {
            main: colors.darkerGreen,
            light: colors.lighterGreen,
            dark: colors.deepGreen,
            contrastText: colors.green,
        },
    },
    shadows: shadows as Shadows,
});
