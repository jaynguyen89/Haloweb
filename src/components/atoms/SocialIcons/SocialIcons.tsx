import { SxProps, Theme } from '@mui/material';
import Box from '@mui/material/Box';
import Grid from '@mui/material/Grid';
import React, { useMemo } from 'react';
import {
    FacebookLogoSvg,
    GoogleLogoSvg,
    InstagramLogoSvg,
    LinkedInLogoSvg,
    MicrosoftMailLogoSvg,
    TwitterLogoSvg,
    AwsLogoSvg,
} from 'src/assets/images';
import _isEmpty from 'lodash/isEmpty';
import useStyles from 'src/components/atoms/SocialIcons/styles';

type TIcon = {
    title: string,
    icon: string,
    handler?: Function,
};

const socialIcons = {
    facebook: {
        title: 'Facebook',
        icon: FacebookLogoSvg,
        handler: undefined,
    } as TIcon,
    google: {
        title: 'Google',
        icon: GoogleLogoSvg,
        handler: undefined,
    },
    twitter: {
        title: 'Twitter',
        icon: TwitterLogoSvg,
        handler: undefined,
    },
    instagram: {
        title: 'Instagram',
        icon: InstagramLogoSvg,
        handler: undefined,
    },
    microsoft: {
        title: 'Microsoft',
        icon: MicrosoftMailLogoSvg,
        handler: undefined,
    },
    linkedin: {
        title: 'Linkedin',
        icon: LinkedInLogoSvg,
        handler: undefined,
    },
    aws: {
        title: 'AWS',
        icon: AwsLogoSvg,
        handler: undefined,
    },
};

export type TSocialIcon = {
    icons: Array<{
        iconName: keyof typeof socialIcons,
        handler?: Function,
    }>,
    variant?: 'inline' | 'grid',
    iconSize?: number,
    sx?: SxProps<Theme>,
};

const SocialIcons = ({
    icons,
    variant = 'grid',
    iconSize = 40,
    sx,
}: TSocialIcon) => {
    const styles = useStyles();

    const selectedIcons = useMemo(() => {
        const filteredIcons = new Array<TIcon>();
        Object.keys(socialIcons).forEach(iconKey => {
            const found = icons.filter(icon => icon.iconName === iconKey);
            if (!_isEmpty(found)) {
                const currentIcon = socialIcons[iconKey as keyof typeof socialIcons];
                currentIcon.handler = found[0].handler;
                filteredIcons.push(currentIcon);
            }
        });

        return filteredIcons;
    }, [icons]);

    const size = useMemo(
        () => variant === 'inline' ? iconSize * 0.75 : iconSize,
        [iconSize, variant],
    );

    if (variant === 'inline') {
        return (
            <Box className={styles.iconBox}>
                {
                    selectedIcons.map(icon =>
                        <Box
                            key={icon.title}
                            className={styles.inlineIcon}
                            sx={sx}
                            component='img'
                            src={icon.icon}
                            width={size}
                            height={size}
                        />)
                }
            </Box>
        );
    }

    return (
        <Grid container spacing={2}>
            {
                selectedIcons.map(icon =>
                    <Grid item sm={2} xs={4} key={icon.title}>
                        <Box
                            className={styles.icon}
                            sx={sx}
                            component='img'
                            src={icon.icon}
                            width={size}
                            height={size}
                        />
                    </Grid>)
            }
        </Grid>
    );
};

export default SocialIcons;
