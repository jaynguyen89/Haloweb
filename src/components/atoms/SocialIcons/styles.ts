import { Theme } from '@mui/material';
import { makeStyles } from '@mui/styles';
import vars from 'src/commons/variables/cssVariables.scss';

const useStyles = makeStyles((theme: Theme) => ({
    iconBox: {
        padding: 0,
        margin: 0,
    },
    icon: {
        cursor: 'pointer',
    },
    inlineIcon: {
        cursor: 'pointer',
        '&:not(:first-child)': {
            marginLeft: `${vars.xtiny} !important`,
        },
    },
}));

export default useStyles;
