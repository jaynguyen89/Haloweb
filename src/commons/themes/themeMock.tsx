import React from 'react';
import { ThemeProvider } from '@mui/material';
import { createTheme } from '@mui/material/styles';
import { render } from '@testing-library/react';
import { shallow, mount } from 'enzyme';

export const MockTheme = ({ children }: { children: React.ReactElement }) => {
    const theme = createTheme({});
    return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
}

export const renderWithTheme = (component: React.ReactElement) => render(
    <MockTheme>{component}</MockTheme>
);

export const shallowWithTheme = (component: React.ReactElement) => shallow(
    <MockTheme>{component}</MockTheme>
);

export const mountWithTheme = (component: React.ReactElement) => mount(
    <MockTheme>{component}</MockTheme>
);
