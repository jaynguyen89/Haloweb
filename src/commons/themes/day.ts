import { createTheme, Shadows } from '@mui/material';
import _cloneDeep from 'lodash/cloneDeep';
import { DEFAULT_THEME_SHADOWS, themeBreakpoints } from 'src/commons/constants';
import colors from 'src/commons/colors';
import { IMixins } from 'src/commons/interfaces';

const shadows = _cloneDeep(DEFAULT_THEME_SHADOWS);
shadows[0] = '0 3px 6px rgba(0,0,0,0.16)';
shadows[1] = '0 3px 6px rgba(0,0,0,0.23)';

export const ThemeDay = createTheme({
    mixins: {
        toolbar: { width: '100%' },
        shadowLight: '0 3px 6px rgba(255,255,255,0.16), 0 3px 6px rgba(255,255,255,0.23)',
        shadowDark: '0 3px 6px rgba(0,0,0,0.16), 0 3px 6px rgba(0,0,0,0.23)',
        shadowLightUp: '0 -3px 6px rgba(255,255,255,0.2)',
        shadowDarkUp: '0px -3px 6px rgba(0, 0, 0, 0.2)',
    } as IMixins,
    breakpoints: { values: themeBreakpoints },
    palette: {
        text: {
            primary: colors.lightBlack,
            secondary: colors.darkGrey,
            disabled: colors.grey,
        },
        primary: {
            main: colors.lightGrey,
            light: colors.lighterGrey,
            dark: colors.darkGrey,
            contrastText: colors.darkerGrey,
        },
        secondary: {
            main: colors.darkTeal,
            light: colors.lightTeal,
            dark: colors.deepTeal,
            contrastText: colors.lighterTeal,
        },
        error: {
            main: colors.darkRed,
            light: colors.lightRed,
            dark: colors.deepRed,
            contrastText: colors.lighterRed,
        },
        warning: {
            main: colors.darkYellow,
            light: colors.lightYellow,
            dark: colors.deepYellow,
            contrastText: colors.lighterYellow,
        },
        info: {
            main: colors.darkBlue,
            light: colors.lightBlue,
            dark: colors.deepBlue,
            contrastText: colors.lighterBlue,
        },
        success: {
            main: colors.darkGreen,
            light: colors.lightGreen,
            dark: colors.deepGreen,
            contrastText: colors.lighterGreen,
        },
        common: {
            white: colors.white,
            black: colors.black,
        },
    },
    shadows: shadows as Shadows,
});
