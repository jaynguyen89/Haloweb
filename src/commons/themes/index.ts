import { ThemeNight } from './night';
import { ThemeDay } from './day';
import { ThemeDream } from './dream';
import { ThemeEarth } from './earth';
import { ThemeGrass } from './grass';
import { ThemeOcean } from './ocean';
import { ThemeRose } from './rose';
import { ThemeSky } from './sky';
import { ThemeSunflower } from './sunflower';
import { ThemeTangerine } from './tangerine';

export default {
    ThemeNight,
    ThemeDay,
    ThemeOcean,
    ThemeDream,
    ThemeSky,
    ThemeGrass,
    ThemeSunflower,
    ThemeTangerine,
    ThemeRose,
    ThemeEarth,
};