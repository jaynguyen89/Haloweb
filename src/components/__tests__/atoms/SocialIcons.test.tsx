import React from 'react';
import { configure, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from '@cfaester/enzyme-adapter-react-18';
import { renderWithTheme } from 'src/commons/themes/themeMock';
import SocialIcons from 'src/components/atoms/SocialIcons/SocialIcons';

const mockHandler = jest.fn() as Function;
const icons = [
    { iconName: 'facebook', handler: mockHandler },
    { iconName: 'google', handler: mockHandler },
    { iconName: 'twitter', handler: mockHandler },
    { iconName: 'instagram', handler: mockHandler },
    { iconName: 'microsoft', handler: mockHandler },
    { iconName: 'linkedin', handler: mockHandler },
    { iconName: 'aws', handler: mockHandler },
];

configure({ adapter: new Adapter() });

describe('SocialIcons', () => {
    it('renders successfully', () => {
        const wrapperGrid = shallow(
            // @ts-ignore-next-line
            <SocialIcons icons={icons} />
        );

        expect(toJson(wrapperGrid)).toMatchSnapshot();

        const wrapperInline = shallow(
            // @ts-ignore-next-line
            <SocialIcons icons={icons} variant='inline' />
        );

        expect(toJson(wrapperInline)).toMatchSnapshot();
    });

    it('is able to render inline and grid layouts', () => {
        const wrapperGrid = renderWithTheme(
            // @ts-ignore-next-line
            <SocialIcons icons={icons} />
        );

        const gridIcons = wrapperGrid.container.getElementsByTagName('img');
        expect(gridIcons.length).toBe(icons.length);

        icons.forEach((entry, i) => {
            expect(gridIcons[i].classList.value.includes('makeStyles-icon')).toBe(true);
            expect(gridIcons[i].classList.value.includes('inlineIcon')).toBe(false);
        });

        const wrapperInline = renderWithTheme(
            // @ts-ignore-next-line
            <SocialIcons icons={icons} variant='inline' />
        );

        const inlineIcons = wrapperInline.container.getElementsByTagName('img');
        expect(inlineIcons.length).toBe(icons.length);

        icons.forEach((entry, i) => {
            expect(inlineIcons[i].classList.value.includes('makeStyles-icon')).toBe(false);
            expect(inlineIcons[i].classList.value.includes('inlineIcon')).toBe(true);
        });
    });
});
