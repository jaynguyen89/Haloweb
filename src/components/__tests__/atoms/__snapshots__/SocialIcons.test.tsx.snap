// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialIcons renders successfully 1`] = `
<ForwardRef(Grid)
  container={true}
  spacing={2}
>
  <ForwardRef(Grid)
    item={true}
    key="Facebook"
    sm={2}
    xs={4}
  >
    <ForwardRef(Box)
      className="makeStyles-icon-2"
      component="img"
      height={40}
      src="logo-facebook.svg"
      width={40}
    />
  </ForwardRef(Grid)>
  <ForwardRef(Grid)
    item={true}
    key="Google"
    sm={2}
    xs={4}
  >
    <ForwardRef(Box)
      className="makeStyles-icon-2"
      component="img"
      height={40}
      src="logo-google.svg"
      width={40}
    />
  </ForwardRef(Grid)>
  <ForwardRef(Grid)
    item={true}
    key="Twitter"
    sm={2}
    xs={4}
  >
    <ForwardRef(Box)
      className="makeStyles-icon-2"
      component="img"
      height={40}
      src="logo-twitter.svg"
      width={40}
    />
  </ForwardRef(Grid)>
  <ForwardRef(Grid)
    item={true}
    key="Instagram"
    sm={2}
    xs={4}
  >
    <ForwardRef(Box)
      className="makeStyles-icon-2"
      component="img"
      height={40}
      src="logo-instagram.svg"
      width={40}
    />
  </ForwardRef(Grid)>
  <ForwardRef(Grid)
    item={true}
    key="Microsoft"
    sm={2}
    xs={4}
  >
    <ForwardRef(Box)
      className="makeStyles-icon-2"
      component="img"
      height={40}
      src="logo-microsoft-exchange.svg"
      width={40}
    />
  </ForwardRef(Grid)>
  <ForwardRef(Grid)
    item={true}
    key="Linkedin"
    sm={2}
    xs={4}
  >
    <ForwardRef(Box)
      className="makeStyles-icon-2"
      component="img"
      height={40}
      src="logo-linkedin.svg"
      width={40}
    />
  </ForwardRef(Grid)>
  <ForwardRef(Grid)
    item={true}
    key="AWS"
    sm={2}
    xs={4}
  >
    <ForwardRef(Box)
      className="makeStyles-icon-2"
      component="img"
      height={40}
      src="logo-aws.svg"
      width={40}
    />
  </ForwardRef(Grid)>
</ForwardRef(Grid)>
`;

exports[`SocialIcons renders successfully 2`] = `
<ForwardRef(Box)
  className="makeStyles-iconBox-1"
>
  <ForwardRef(Box)
    className="makeStyles-inlineIcon-3"
    component="img"
    height={30}
    key="Facebook"
    src="logo-facebook.svg"
    width={30}
  />
  <ForwardRef(Box)
    className="makeStyles-inlineIcon-3"
    component="img"
    height={30}
    key="Google"
    src="logo-google.svg"
    width={30}
  />
  <ForwardRef(Box)
    className="makeStyles-inlineIcon-3"
    component="img"
    height={30}
    key="Twitter"
    src="logo-twitter.svg"
    width={30}
  />
  <ForwardRef(Box)
    className="makeStyles-inlineIcon-3"
    component="img"
    height={30}
    key="Instagram"
    src="logo-instagram.svg"
    width={30}
  />
  <ForwardRef(Box)
    className="makeStyles-inlineIcon-3"
    component="img"
    height={30}
    key="Microsoft"
    src="logo-microsoft-exchange.svg"
    width={30}
  />
  <ForwardRef(Box)
    className="makeStyles-inlineIcon-3"
    component="img"
    height={30}
    key="Linkedin"
    src="logo-linkedin.svg"
    width={30}
  />
  <ForwardRef(Box)
    className="makeStyles-inlineIcon-3"
    component="img"
    height={30}
    key="AWS"
    src="logo-aws.svg"
    width={30}
  />
</ForwardRef(Box)>
`;
